ocaml/sdk-gen/go/templates/Option.mustache

```diff
@@ -0,0 +1,4 @@
+{{#option}}
+type Option{{func_suffix}} *{{type}}
```

Comment:
Could you please add a `options.go` as a test for this template?

Reply:
`func_suffix` -> `type_name_suffix`

Reply:
I have added rendered test for `option`. For the result is only a line, I put it in the code:
```ocaml
let option_rendered = "type OptionString *string"

  let tests =
    `QuickAndAutoDocumented
      [
        ......
      ; (("Option.mustache", option), option_rendered)
      ]
```

Reply:
I can't see how this will be handled by templates.
`string_of_ty_with_enums` needs to return an indicator for templates to render the `Option` types specially. In this case, the `Option` types will be converted to pointers. How does a template know that w/o the indicator?

Reply:
Great. It does leave out the handling of options in `string_of_ty_with_enums`.